src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -80,7 +98,17 @@ internal void Return(T[] buffer)
                 // we can just drop this buffer
                 if (_index != 0)
                 {
+                    // If we have a debugger attached, do an extra safety check
+                    // to prevent a double-return scenario
+                    if (System.Diagnostics.Debugger.IsAttached)
```

Comment:
IIRC IsAttached is cheap, but if you haven't already you might want to double-check that this call doesn't have adverse affects on perf.